### PR TITLE
Fix: `indent` rule for multi-line objects and arrays (fixes #3236)

### LINKED ISSUE #3236: Indent rule problem with object and array literals
Version 1.0.0

Rule:
```yml
indent: [2, 2, {"SwitchCase": 1, "VariableDeclarator": 2}]
```

Code:
```js
var testCases =
    [{ crl: 'ca2-crl',
      clients:
       [
        { name: 'agent1', shouldReject: true, shouldAuth: false },
        { name: 'agent2', shouldReject: true, shouldAuth: false }
       ]
    }
    ];

var abc =
     {
       x: 2,
        y: 4
     };
```

Result:
```
test1.js
   5:9  error  Expected indentation of 9 characters but found 8  indent
   6:9  error  Expected indentation of 9 characters but found 8  indent
  14:9  error  Expected indentation of 7 characters but found 8  indent
```

The rule accepts any indentation for the object or array literal if it starts on a new line. Then it uses this indentation as a reference for the next errors, so we see an expected odd indentation (9, 7) when the rule is set to 2.

Cross-reference: https://github.com/nodejs/io.js/pull/2286

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: lib/options.js [hunk 1]
```diff
@@ -19,109 +19,122 @@ module.exports = optionator({
...
-        type: "String",
-        description: "Specify filename to process STDIN as"
-    },
-    {
-        option: "init",
-        type: "Boolean",
-        default: "false",
-        description: "Run config initialization wizard"
-    }]
+    options: [
```
Comment (nzakas) [inline lib/options.js]: Did the previous format trigger an indent warning? If so, then we should make sure it doesn't, as the previous format should also be valid.

Comment (gyandeeps) [inline lib/options.js]: I think it should be a warning as the content inside an array should have an indent.

Comment (nzakas) [inline lib/options.js]: I'm saying we can't force people to do that. There's no reason to assume array literals must always have an indent, especially if formatted the way this section used to be.

Comment (gyandeeps) [inline lib/options.js]: If I understand it correctly, you are saying the following is valid:

```js
var abc = [
{
    a: 1
},
{
    b: 2
}
];
```

Comment (nzakas) [inline lib/options.js]: No, I'm saying this is valid:

```js
abc = [{
    a: 1
},
{
    b: 2
}];
```

So if the open bracket is not immediately followed by a newline, then the indentation level should not be affected.

Comment (gyandeeps) [inline lib/options.js]: Got it. Will make that change.

**Comment by gyandeeps** [comment]
@nzakas Done.

**Pull request merged**